src/structure/pdb.jl

```diff
@@ -30,26 +49,326 @@ end
 
 
 """
-Download a Protein Data Bank (PDB) file or biological assembly from the RCSB
-PDB. By default downloads the PDB file; if `ba_number` is set the biological
-assembly with that number will be downloaded.
+Returns a list of all PDB entries from RCSB PDB server.
 """
-function downloadpdb(pdbid::AbstractString,
-                    out_filepath::AbstractString="$pdbid.pdb";
-                    ba_number::Integer=0)
+function pdbentrylist()
+    pdbidlist = String[]
+    info("Fetching list of all PDB Entries from RCSB PDB Server...")
+    tempfilepath = tempname()
+    download("ftp://ftp.wwpdb.org/pub/pdb/derived_data/index/entries.idx",tempfilepath)
+    open(tempfilepath) do input
+        # Skips the first two lines as it contains headers
+        linecount = 1
+        for line in eachline(input)
+            if linecount > 2
+                # The first 4 characters in the line is the PDB ID
+                pdbid = uppercase(line[1:4])
+                # Check PDB ID is 4 characters long and only consits of alphanumeric characters
+                if length(pdbid) != 4 || ismatch(r"[^a-zA-Z0-9]", pdbid)
+                    throw(ArgumentError("Not a valid PDB ID: \"$pdbid\""))
+                end
+                push!(pdbidlist,pdbid)
+            end
+            linecount +=1
+        end
+    end
+    rm(tempfilepath)
```

Comment:
`rm(tempfilepath, force=true)` would be better because it works even when the file does not exist.